### ISSUE TITLE
[Release request] Release v3 minor version with atomic styled support

### ISSUE THREAD

**Issue by yepitschunked** [callstack/linaria#1174]
## Describe the feature

Release a v3.1 (or whatever minor version is preferred) containing the `styled` atomic support in https://github.com/callstack/linaria/pull/966, without the changes in 3.0.0-beta.19. 

## Motivation

I work with @jpnelson and we're currently running Linaria with atomic styles on 3.0.0-beta.18, which has been stable and working well for us over the past 6 months or so. We'd like to use the `styled` atomic support contributed by @juanferreras in https://github.com/callstack/linaria/pull/966, but it was never published to NPM. Instead, 3.0.0-beta.19 was released, which contains a major refactor to how tags are resolved, and also removes the `libResolver` option[^1] that was contributed in https://github.com/callstack/linaria/issues/851. v4 contains even bigger changes and we've run into several issues with shaker/preeval on our codebase, so we'd like to hold off on those upgrades for now.

I'll be putting up several PRs for issues we found while testing a v4 upgrade, so hopefully we can get up to date later this year! 

[^1]: We're using libResolver as a way to enable atomic styles across our codebase without needing to make code changes (via swapping `@linaria/core` and `@linaria/atomic` imports). 

## Possible implementations

Cut a release at this revision: https://github.com/callstack/linaria/commit/f59860b09c5f91b0423dbf188e5f8aaaef38a6b5

**Comment by jpnelson** [comment]
Yeah, I think it'd make sense too given that v3 has partial atomic support already – it's not really a complete solution without `styled` support too.

@Anber what do you think about this? We'd still like to get onto v4 but need to figure out those blockers mentioned first.

**Comment by Anber** [comment]
Hi @yepitschunked @jpnelson!
I've published [f38a6b5](https://github.com/callstack/linaria/commit/f59860b09c5f91b0423dbf188e5f8aaaef38a6b5) under the `legacy` tag.
